src/DateTimePickerAndroid.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @format
+ * @flow strict-local
+ */
+import {
+  DATE_SET_ACTION,
+  TIME_SET_ACTION,
+  DISMISS_ACTION,
+  NEUTRAL_BUTTON_ACTION,
+  ANDROID_DISPLAY,
+  ANDROID_MODE,
+  ANDROID_EVT_TYPE,
+  EVENT_TYPE_SET,
+} from './constants';
+import invariant from 'invariant';
+
+import type {DateTimePickerEvent, AndroidNativeProps} from './types';
+import {
+  getOpenPicker,
+  timeZoneOffsetDateSetter,
+  validateAndroidProps,
+} from './androidUtils';
+import pickers from './picker';
+
+function open(props: AndroidNativeProps) {
+  const {
+    mode = ANDROID_MODE.date,
+    display = ANDROID_DISPLAY.default,
+    value: originalValue,
+    is24Hour,
+    minimumDate,
+    maximumDate,
+    neutralButtonLabel,
+    minuteInterval,
+    timeZoneOffsetInMinutes,
+    onChange,
+    onError,
+  } = props;
+  validateAndroidProps(props);
+  invariant(originalValue, 'A date or time must be specified as `value` prop.');
+
+  const valueTimestamp = originalValue.getTime();
+  const openPicker = getOpenPicker({
+    mode,
+    value: valueTimestamp,
+    display,
+    is24Hour,
+    minimumDate,
+    maximumDate,
+    neutralButtonLabel,
+    minuteInterval,
+    timeZoneOffsetInMinutes,
+  });
+
+  const presentPicker = async () => {
+    try {
+      const {action, day, month, year, minute, hour} = await openPicker();
+      let date = new Date(valueTimestamp);
+      let event: DateTimePickerEvent = {
+        type: EVENT_TYPE_SET,
+        nativeEvent: {},
+      };
+
+      switch (action) {
+        case DATE_SET_ACTION:
+          date.setFullYear(year, month, day);
+          date = timeZoneOffsetDateSetter(date, timeZoneOffsetInMinutes);
+          event.nativeEvent.timestamp = date.getTime();
+          onChange?.(event, date);
+          break;
+
+        case TIME_SET_ACTION:
+          date.setHours(hour, minute);
+          date = timeZoneOffsetDateSetter(date, timeZoneOffsetInMinutes);
+          event.nativeEvent.timestamp = date.getTime();
+          onChange?.(event, date);
+          break;
+
+        case NEUTRAL_BUTTON_ACTION:
+          event.type = ANDROID_EVT_TYPE.neutralButtonPressed;
+          onChange?.(event, originalValue);
+          break;
+
+        case DISMISS_ACTION:
+        default:
+          event.type = ANDROID_EVT_TYPE.dismissed;
+          onChange?.(event, originalValue);
+          break;
+      }
+    } catch (error) {
+      onError && onError(error);
+    }
+  };
+  presentPicker();
+}
+
+function dismiss(mode: AndroidNativeProps['mode']) {
```

Comment:
This syntax seem to be invalid with older versions of RN/flow (tested on RN 63.3/flow 0.122.0):

```
error: SyntaxError: .../node_modules/@react-native-community/datetimepicker/src/DateTimePickerAndroid.js: Unexpected token, expected "]" (97:42)

   95 | }
   96 |
>  97 | function dismiss(mode: AndroidNativeProps['mode']) {
      |                                           ^
   98 |   // $FlowFixMe - `AbstractComponent` [1] is not an instance type.
   99 |   pickers[mode].dismiss();
  100 | }
```

Reply:
hello, yes, you'll need Flow 155: https://flow.org/en/docs/types/indexed-access/

Reply:
We are not using flow in project RN 0.63.4 and having this issue. Not sure how to fix. Do you have a workaround?
Tried to use older versions which is 5.1.0 and 3.5.2 but for those i am getting "Unhandled promise rejection [TypeError: Cannot read property 'open' of null" any idea or guidence to help on this?